ports/esp32/esp32_rmt.c

```diff
@@ -121,14 +133,11 @@ STATIC mp_obj_t esp32_rmt_make_new(const mp_obj_type_t *type, size_t n_args, siz
 STATIC void esp32_rmt_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
     esp32_rmt_obj_t *self = MP_OBJ_TO_PTR(self_in);
     if (self->pin != -1) {
-        mp_printf(print, "RMT(channel=%u, pin=%u, source_freq=%u, clock_div=%u",
-            self->channel_id, self->pin, APB_CLK_FREQ, self->clock_div);
-        if (self->carrier_freq > 0) {
-            mp_printf(print, ", carrier_freq=%u, carrier_duty_percent=%u)",
-                self->carrier_freq, self->carrier_duty_percent);
-        } else {
-            mp_printf(print, ")");
-        }
+        bool idle_output_en;
+        rmt_idle_level_t idle_level;
+        check_esp_err(rmt_get_idle_level(self->channel_id, &idle_output_en, &idle_level));
+        mp_printf(print, "RMT(channel=%u, pin=%u, source_freq=%u, clock_div=%u, idle_level=%u)",
+            self->channel_id, self->pin, APB_CLK_FREQ, self->clock_div, idle_level);
```

Comment:
There's no API for retrieving the TX carrier state once you've set it and I wasn't convinced it was worth wasting the bytes in the object to store it just so it can be shown in the print function. However, I'm happy to do this if you think so.

Reply:
Ah ok.  Just leave it as-is for now (ie don't print the TX carrier state).  Can be added later if it's really needed.